pkg/controller/controller.go

```diff
@@ -32,13 +33,19 @@ type ControllerFunc func() error
 // ExitReason is a returnable type from DoFunc that causes the
 // controller to exit. This reason is recorded in the controller's status. The
 // controller is not removed from any manager.
-// Construct one with ExitReason{errors.New("a reason")}
+// Construct one with MakeExitReason("a reason")
 type ExitReason struct {
 	// This is constucted in this odd way because the type assertion in
 	// runController didn't work otherwise.
 	error
 }
 
+// MakeExitReason constructs an error object that, when returned by a DoFunc,
+// stops the controller from running further.
+func MakeExitReason(msg string) ExitReason {
```

Comment:
Unit test this please, and plumb its usage wherever creation of an exit reason is called directly in the controller code.